pkg/api/pod/util.go

```diff
@@ -302,3 +312,11 @@ func DropDisabledVolumeDevicesAlphaFields(podSpec *api.PodSpec) {
 		}
 	}
 }
+
+// DropDisabledAlphaStatusFields removes disabled fields from the pod status.
+// This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a pod status.
```

Comment:
It's because PodStatus is discarded in podStrategy's PrepareForCreate and set to a default PodStatus which contains no alpha fields.

This comment made me realize that I was using DropDisabledAlphaStatusFields inconsistently in the multiple PrepareForUpdates, though, so I added it to all of them.

Reply:
I think we can add it in `PrepareForCreate` as well so that this doesn't break in the future.

Reply:
Makes sense, done.